algorithms/searching/binary_search.py

```diff
@@ -0,0 +1,22 @@
+# Binary Search
+'''
+This Works on a Sorted Array.
+Time Complexity => O(log n)
+'''
+# Solved Using Recursion.
+def binary(a, first, last, term):
+    mid=int((first+last)/2)
+    if term>a[mid]:
+        binary(a, mid, last, term)
+    elif term<a[mid]:
+        binary(a,first, mid, term)
+    elif term==a[mid]:
+        print("Number found at", mid+1)
+    else:
+        print("Number is not there in the array")
+
```

Comment:
Please follow Pep8 and make the binary function iterative. Otherwise, everything else LGTM :+1: